src/modules/kvs/kvs.c

```diff
@@ -2726,24 +2762,79 @@ static int checkpoint_get (flux_t *h, const char *key, char *buf, size_t len)
         return -1;
     if (flux_rpc_get_unpack (f, "{s:s}", "value", &value) < 0)
         goto error;
-    if (strlen (value) >= len) {
+
+    /* if value is a blobref, its verison 0 checkpoint */
+    if (blobref_validate (value) == 0)
+        return checkpoint_get_version0 (h, value, buf, len);
+
+    if (!(o = json_loads (value, 0, NULL))) {
         errno = EINVAL;
         goto error;
     }
-    strcpy (buf, value);
-    flux_future_destroy (f);
```

Comment:
With above changes, this can just be
```c
if (checkpoint_get_version1() < 0 && checkpoint_get_version0() < 0 {
    // error interpreting checkpoint object
}
```